shopinvader/models/shopinvader_variant.py

```diff
@@ -176,60 +176,97 @@ def _compute_variant_attributes(self):
+        """Computes the product prices
 
-    def _get_price_per_qty(self, qty, pricelist, fposition, company=None):
-        product_id = self.record_id
-        taxes = product_id.taxes_id.sudo().filtered(
-            lambda r: not company or r.company_id == company
-        )
-        # get the expeced tax to apply from the fiscal position
-        tax_id = fposition.map_tax(taxes, product_id) if fposition else taxes
-        tax_id = tax_id and tax_id[0]
-        product = product_id.with_context(
-            quantity=qty, pricelist=pricelist.id, fiscal_position=fposition
-        )
-        final_price, rule_id = pricelist.get_product_price_rule(
-            product, qty or 1.0, None
+        :param qty:         The product quantity, used to apply pricelist rules.
+        :param pricelist:   Optional. Get prices for a specific pricelist.
+        :param fposition:   Optional. Apply fiscal position to product taxes.
+        :param company:     Optional.
+
+        :returns: dict with the following keys:
+
+            <value>                 The product unitary price
+            <tax_included>          True if product taxes are included in <price>.
+
+            If the pricelist.discount_policy is "without_discount":
+            <original_value>        The original price (before pricelist is applied).
+            <discount>              The discounted percentage.
+        """
+        self.ensure_one()
+        AccountTax = self.env["account.tax"]
+        product = self.record_id
+        # Apply company
+        product = product.with_company(company) if company else product
+        company = company or self.env.company
+        # Always filter taxes by the company
+        taxes = product.taxes_id.filtered(lambda tax: tax.company_id == company)
+        # Apply fiscal position
+        taxes = fposition.map_tax(taxes, product) if fposition else taxes
+        # Set context. Some of the methods used here depend on these values
+        product_context = dict(
+            self.env.context,
+            quantity=qty,
+            pricelist=pricelist.id if pricelist else None,
+            fiscal_position=fposition,
         )
-        tax_included = tax_id.price_include
-        account_tax_obj = self.env["account.tax"]
-        # fix tax on the price
-        value = account_tax_obj._fix_tax_included_price_company(
-            final_price, product.taxes_id, tax_id, company
+        product = product.with_context(**product_context)
+        pricelist = pricelist.with_context(**product_context) if pricelist else None
+        # If we have a pricelist, use product.price as it already accounts
+        # for pricelist rules and quantity (in context)
+        price_unit = product.price if pricelist else product.lst_price
+        price_unit = AccountTax._fix_tax_included_price_company(
+            price_unit, product.taxes_id, taxes, company
         )
+        # TODO: During next migration, rename value and original_value to
```

Comment:
Note we call the variable value and original_value because on the frontend this dict belong to the key "price"

example of product json

```
{
    "name": "FOO",
    "price": {
        "value": 10,
        "tax_included": True,
        "original_value": 15,
        "discount": 30%,
    }
}
```

So in frontend template getting the price is something like
```
product.price.value
```

renaming value by price will give something like

```
product.price.price
```

So I think we should always keep the current naming

Reply:
I see, makes sense. I'll remove the comment